release/models/policy-forwarding/openconfig-policy-forwarding.yang

```diff
@@ -62,11 +62,14 @@ module openconfig-policy-forwarding {
 
     A forwarding-policy specifies the match criteria that it intends
     to use to determine the packets that it reroutes - this may
-    consist of a number of criteria, such as DSCP. The action of the
-    policy results in a forwarding action being applied to matching
-    packets. For example, decapsulating the packet from a GRE header.
-    In order to enact the policy based on particular interfaces - the
-    forwarding-policy is applied to an interface via referencing it
+    consist of a number of criteria, such as DSCP. The match criteria
+    is specified as rules.  If no rules are specified, then the policy
+    will match all packets.
```

Comment:
Why is this the right thing to do vs. having an explicit match-all type criteria?

The downside of this API design is that the absence of something means something implicitly -- it seems a better API design to have the presence of something indicate something, which would mean that we should rather have an explicit "match everything" (can be done with matching all src/dst IP addrs)?

Reply:
There is no right or wrong here.   This proposal is made based on precedent as referenced in  https://github.com/openconfig/public/pull/1223#issuecomment-2501782886 earlier in this PR.  The industry precedent is no-condition = matches any and all.  This was also discussed in the OpenConfig operator meeting on Nov 26, 2024 with some thoughtful conversation.    

The precedent is pretty strong where 5 out of 5 implement that no "condition" means match all.    (Not included in the gist of 4 examples is arista EOS, which also defines a [route map with no match statements as match all](https://www.arista.com/en/um-eos/eos-acls-and-route-maps))

I do like the idea of being explicit in general.  A number of changes in OC in the last few years have been due to ambiguity.  But what is better?  Do something different in OC or go with the precedent?   In this case, I think the precedent is so strong that it's better to go with no conditions means match all.  This change makes this decision explicit via description, but not in the data model/API.